driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/ChangeStreamPublisherImpl.java

```diff
@@ -112,12 +112,18 @@ private ChangeStreamPublisherImpl(
 
     @Override
     public <TDocument> Publisher<TDocument> withDocumentClass(final Class<TDocument> clazz) {
-        return new BatchCursorPublisher<TDocument>(getClientSession(), getMongoOperationPublisher().withDocumentClass(clazz)) {
+        BatchCursorPublisher<TDocument> result = new BatchCursorPublisher<TDocument>(
```

Comment:
> Depending on how the ChangeStreamPublisher is used a explicitly declared batchSize maybe lost. If its set before withDocumentClass is called its lost, if its set after its ok. If its lost it will be set the signalled demand value.

Was this meant to be the opposite?

- It seems to me that if `ChangeStreamPublisherImpl.batchSize` is called before `ChangeStreamPublisherImpl.withDocumentClass`, then with the changes in this PR, the specified `batchSize` is used by the new `BatchCursorPublisher`, i.e., it is not lost.
- If `ChangeStreamPublisherImpl.batchSize` is called after `ChangeStreamPublisherImpl.withDocumentClass`, then the `BatchCursorPublisher` created by the method `withDocumentClass` cannot observe the new `batchSize`, and it is lost. I do not see how adding a new constructor to `BatchCursorPublisher` would change this, am I missing something?

Reply:
I just meant you can clean up the code by just adding a new constructor eg: rozza@50b51d6

Reply:
> Was this meant to be the opposite?
I was describing why this fix was needed after Jeff asked _"It's not clear to me why batchSize is the one property that isn't propagated. Can you explain what's going on?"_

As `withDocumentClass` returns a plan Publisher `batchSize` cannot be set after conversion. Batchsize was only implicitly set via demand, even if a user explicitly set it by calling `batchSize` before calling `withDocumentClass`.

So this fix, fixes that niche case where a user wants to control the batchSizes from the cursor and use an alternative document class.

Reply:
Thanks, @rozza, I merged your proposed commit. I still do not completely understand your explanation, likely because I lack more broad understanding of how this class is used.